### PR TITLE
Fix config, make sure timer flush fast path is working

### PR DESCRIPTION
**Pull request by cw9**
@xichen2020

### CODE REVIEW

File: aggregator/elem.go [hunk 1]
```diff
@@ -359,16 +359,16 @@ func NewTimerElem(id id.RawID, sp policy.StoragePolicy, aggTypes policy.Aggregat
 
 // ResetSetData overrides elemBase.ResetData, to include quantile update for timer elements.
 func (e *TimerElem) ResetSetData(id id.RawID, sp policy.StoragePolicy, aggTypes policy.AggregationTypes) {
+	e.elemBase.ResetSetData(id, sp, aggTypes)
+
 	if aggTypes.IsDefault() {
-		aggTypes = e.opts.DefaultTimerAggregationTypes()
+		e.aggOpts.ResetSetData(e.opts.DefaultTimerAggregationTypes())
```
Comment (xichen2020) [inline aggregator/elem.go]: Put a TODO in there to ensure the aggTypes stored in elemBase agree with what we store in aggregation options?

File: config/m3aggregator.yaml [hunk 2]
```diff
@@ -76,10 +77,6 @@ aggregator:
           capacity: 32
         - count: 1024
           capacity: 64
-  timerQuantiles:
```
Comment (xichen2020) [inline config/m3aggregator.yaml]: Can we keep the timer quantiles? I'd rather keep them explicit if possible

File: config/m3aggregator.yaml [hunk 3]
```diff
@@ -61,6 +61,7 @@ http:
   writeTimeout: 45s
 
 aggregator:
+  timerAggregationTypes: Sum,SumSq,Mean,Lower,Upper,Count,Stdev,Median,P50,P95,P99
```
Comment (xichen2020) [inline config/m3aggregator.yaml]: Add a comment here saying they need to be comma deliminated string? Might be more consistent to make it a list like the quantiles, though

File: aggregator/elem.go (hunk 1 above)
Comment (cw9) [inline aggregator/elem.go]: Sure

File: config/m3aggregator.yaml (hunk 3 above)
Comment (cw9) [inline config/m3aggregator.yaml]: Sure, we could change it to a list of agg types eventually

File: config/m3aggregator.yaml (hunk 2 above)
Comment (cw9) [inline config/m3aggregator.yaml]: We can't, as we don't support setting timerquantiles in the options now, and quantiles needs to be valid aggregation types anyway so we should just set aggregation types

File: config/m3aggregator.yaml (hunk 3 above)
Comment (xichen2020) [inline config/m3aggregator.yaml]: Yeah can we do that in this diff? A list of types looks much easier on the eyes

File: config/m3aggregator.yaml (hunk 2 above)
Comment (xichen2020) [inline config/m3aggregator.yaml]: Right yeah makes sense

**Comment by xichen2020** [comment]
Also do we have a regression test that catches the bug you fixed? If not, can we add it?

**Comment by cw9** [comment]
I added a regression test in elem_test.go

**Comment by cw9** [comment]
You need to stamp this diff for the yaml parsing to work: https://github.com/m3db/m3metrics/pull/53

**Comment by xichen2020** [comment]
Yeah stamped the other diff.

File: config/m3aggregator.yaml (hunk 3 above)
Comment (cw9) [inline config/m3aggregator.yaml]: done

**Pull request merged**